source/common/upstream/subset_lb.cc

```diff
@@ -330,6 +333,21 @@ bool SubsetLoadBalancer::hostMatches(const SubsetMetadata& kvs, const Host& host
       return false;
     }
 
+    // If configured to do so, we attempt to match the metadata value with the contents
+    // of the list specified on the host.
+    if (list_as_any_) {
+      bool list_match = false;
+      for (const auto& l : entry_it->second.list_value().values()) {
+        if (ValueUtil::equal(l, kv.second)) {
+          list_match = true;
+        }
```

Comment:
Add a break to skip the rest of the list? Could even just directly continue the outer loop and skip tracking list_match (though using list_match may be easier to read).